js/src/vue/app.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+import { createApp } from 'vue/dist/vue.esm-bundler.js';
+import {defineAsyncComponent} from "vue";
+window.Vue = {
+    createApp: createApp,
+    components: {}
+};
+// Require all Vue SFCs in js/src directory
+const component_context = import.meta.webpackContext('./..', {
+    regExp: /\.vue$/i,
+    recursive: true,
+    mode: 'lazy',
+    chunkName: '/vue-sfc/[request]'
+});
+const components = {};
+component_context.keys().forEach((f) => {
+    // Ex: ./Debug/Toolbar.vue => DebugToolbar
+    const component_name = f
+        .replace(/\/([a-z])/g, (match, p1) => p1.toUpperCase())
+        .replace(/\.vue$/, '')
+        .replace(/\//g, '')
+        .replace(/[^a-zA-Z0-9]/g, '');
```

Comment:
It seems that it can work even by keeping the `/`. I was able to display the debug bar (not widgets, but I guess some code has to be updated).

Also, keeping the `/` in directory structure could prevent to have to define a "tagging" logic here, that seems to be used to differenciate `Debug/Widget/*` components and `Debug/WidgetButton` component. If we keep the directory structure, then `window.Vue.getComponentsByTag('debug-widget')` could be replaced by a `window.Vue.getComponentsByName(new RegExp('/^Debug\\/Widget\\//'))`, and current file will not have to implement any tagging directive.
If we want to keep a tagging logic, we have to a solution that assign tags using directives defined by the component themselves. Indeed, it would probably a pain to maintain if the current file have the responsibility to define components tags.

```suggestion
    // Ex: ./Debug/Toolbar.vue => Debug/Toolbar
    const component_name = f.replace(/^\.\/(.+)\.vue$/, '$1');
```